### DISCUSSION TITLE
Documentation on pdm lock

### DISCUSSION THREAD

**Discussion by yingbo** [pdm-project/pdm#1029] [Q&A] [answered]
## Is your feature request related to a problem? Please describe.

I don't understand what's `pdm lock` doing, and when to use `pdm lock` or `pdm update`. 

## Describe the solution you'd like

Hope have a clear explanation. Thanks a lot!

**Comment by pawamoy** [comment]
Locking is resolving the whole dependencies set again, without using any prior resolution data.
Updating will reuse resolution data if possible. Different update strategies are available, see https://pdm.fming.dev/usage/dependency/#update-existing-dependencies. It means that updating can be faster than locking (given you already resolved dependencies at least once).

**Comment by pawamoy** [comment]
I'll move this to discussions as it's not an issue.

**Reply by yingbo** [reply]
Thanks. Sorry I didn't know we have a discussion place.

**Comment by frostming** [comment, marked as answer]
Lock file will be created automatically if not exists when you run `pdm install`. You only need to run `pdm lock` when you believe the lock file is corrupted or become unusable due to an upgrade.

With the existence of a lock file, `pdm install` and `pdm sync` will only install versions from it. So if you want the versions to be updated, use `pdm update`, it can be followed by package names so only selected package versions will be updated.
